src/Agent.Sdk/Util/PlatformUtil.cs

```diff
@@ -152,13 +152,32 @@ private static void DetectRHEL6()
             }
         }
 
+        private static string GetLinuxReleaseFilePath()
+        {
+            if (RunningOnLinux)
+            {
+                if (File.Exists("/etc/os-release"))
+                {
+                    return "/etc/os-release";
+                }
+                else if (File.Exists("/usr/lib/os-release"))
+                {
+                    return "/usr/lib/os-release";
+                }
+            }
+
+            return null;
+        }
+
         private static string GetLinuxId()
         {
-            if (RunningOnLinux && File.Exists("/etc/os-release"))
+
+            if (RunningOnLinux)
             {
+                string filePath = GetLinuxReleaseFilePath();
                 Regex linuxIdRegex = new Regex("^ID\\s*=\\s*\"?(?<id>[0-9a-z._-]+)\"?");
 
-                using (StreamReader reader = new StreamReader("/etc/os-release"))
+                using (StreamReader reader = new StreamReader(filePath))
```

Comment:
It's a bit confusing why we try open stream with null here. Can we just make a null check before?
If we expect the stream may fail, will it be the same exception message?